### ISSUE TITLE
Argo-workflows helm chart should restart deployment on configmap change

### ISSUE THREAD

**Issue by brian-bk** [argoproj/argo-helm#2097]
### Describe the bug

I found that when configuration is made to change the configmap, this does not trigger a restart of the controller deployment. It would be nice if pods using the stale configmap would become tainted.

I've seen clever ways to taint pods in helm charts, like:

```
annotations:
  checksum/controller-cm: {{ include (print $.Template.BasePath "/workflow-controller-config-map.yaml") . | sha256sum | quote }}
```

In the meantime, a `kubectl rollout restart deploy/{controller} deploy/{server}` works!

### Related helm chart

argo-workflows

### Helm chart version

0.20.6

### To Reproduce

* Deploy argo-workflows helm chart by argocd
* Update some configuration, but not the pod annotations, that will update the configmap
* Argo-workflows still leverages old configuration indefinitely, until...
* Deployment is manually restarted

### Expected behavior

* Deploy argo-workflows helm chart by argocd
* Update some configuration, but not the pod annotations, that will update the configmap
* Deployment is automatically restarted

**Comment by yu-croco** [comment]
Hi @brian-bk ,
This is not an issue of Helm Chart but upstream's ( https://github.com/argoproj/argo-workflows/issues/10659 ).
This is fixed on Argo Workflows v3.4.6 .

**Comment by brian-bk** [comment]
Ah thanks! Feel free to close.
